### PR TITLE
Add README to couchapps

### PR DESCRIPTION
**Pull request by esoergel**
@orangejenny @gcapalbo
pretty view: https://github.com/dimagi/commcare-hq/blob/couchapps-docs/corehq/couchapps/README.md

### CODE REVIEW

**Comment by esoergel** [comment]
@dannyroberts, you might also be interested.

File: corehq/couchapps/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+With this new structure, you can make a change to `"app_builds_by_date"`
+without needing to also reindex `"saved_apps"` and `"forms_by_xmlns"`.
+
+In the example above, the `forms_by_xmlns` view operates on XFormInstance, but
+for some reason lives in `app_manager`.  By moving it here, you can explicitly
+choose in which dbs to include that view.
+
+Writing `Application.view("app_manager/forms_by_xmlns", ...)` should seem a
+little weird to you, because it makes sense for views to be associated with the
+doc type they operate on, not the module (if the two disagree).
```
Comment (benrudolph) [inline corehq/couchapps/README.md]: nice! really dig this and makes sense. :beers:

**Comment by czue** [comment]
Nice! :book: :books: :blue_book:

**Comment by snopoke** [comment]
Great docs :book:

**Pull request merged**